WalletWasabi/Crypto/Mac.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Text;
+using NBitcoin.Secp256k1;
+using WalletWasabi.Crypto.Groups;
+using WalletWasabi.Helpers;
+
+namespace WalletWasabi.Crypto
+{
+	public class MAC : IEquatable<MAC>
+	{
+		private MAC(Scalar t, GroupElement V)
+		{
+			this.t = CryptoGuard.NotZero(nameof(t), t);
+			this.V = CryptoGuard.NotNullOrInfinity(nameof(V), V);
+		}
+
+		public Scalar t { get; }
+		public GroupElement V { get; }
+
+		public static bool operator == (MAC a, MAC b) => a.Equals(b);
+		public static bool operator != (MAC a, MAC b) => !a.Equals(b);
+
+		public bool Equals(MAC? other) =>
+			(this, other) switch
+			{
+				(null, null) => true,
+				(null, {}) => false,
+				({}, null) => false,
+				({} a , {} b) when Object.ReferenceEquals(a, b) => true, 
+				({} a , {} b) => (a.t, a.V) == (b.t, b.V)
+			};
+
+		public override bool Equals(object? obj) =>
+			Equals(obj as MAC);
+
+		public override int GetHashCode() =>
+			HashCode.Combine(t, V).GetHashCode();
+
+		public static MAC ComputeMAC(CoordinatorSecretKey sk, GroupElement Ma, Scalar t)
+		{
+			Guard.NotNull(nameof(sk), sk);
+			CryptoGuard.NotNullOrInfinity(nameof(Ma), Ma);
```

Comment:
if we ensure `Ma` is not infinity, i guess it makes sense to ensure `t` isn't `Zero`?

Reply:
thinking about it more, it's not just for consistency, this is actually important for the security proofs, but maybe it's not important for our use case. we should do it anyway. it never makes sense to do allow `t` to be 0 because that cancels out the `x1` component of the secret key.

in principle it could make sense to generate a MAC over a group element that is the point at infinity, but not in wabisabi so we should keep that restriction.

Reply:
done